include/roboptim/core/function/polynomial.hh

```diff
@@ -72,6 +72,16 @@ namespace roboptim
     value_type applyPolynomial
     (const_vector_ref coeffs, const_argument_ref x) const;
 
+  public:
+    /// \brief Get the type-checking flag
+    virtual unsigned long flag()
+    {
+      return flag_;
+    }
+
+    /// \brief Flag representing the Roboptim Function type
+    static const unsigned long flag_;
+
```

Comment:
Wouldn't it be better to move this into a macro if it needs to be added to each and every function? Also, `unsigned long` could be changed by a typedef, in case this needs to be changed later on (e.g. number of flags increases because RobOptim becomes too mainstream!).